include/auth.php

```diff
@@ -100,8 +100,9 @@ function identifyGroups() {
 				if (in_array(strtolower($this->user), preg_split('/\s*,\s*/', $names))) {
 					$this->usersGroups[] = '@'.$group;
 				}
-
-				foreach ($this->usersGroups as $users_group) {
+			}
+			foreach ($this->usersGroups as $users_group) {
+				foreach ($groups as $group => $names) {
 					if (in_array($users_group, preg_split('/\s*,\s*/', $names))) {
```

Comment:
```suggestion
					if (in_array($users_group, preg_split('/\s*,\s*/', $names)) && !in_array($this->usersGroups, '@'.$group)) {
```
should fix any issues from circular groups

Reply:
your in_array syntax is wrong, but I have added this